### PR TITLE
fix: now hub knows bid_id for deal

### PR DESCRIPTION
**Pull request by sshaman1101**

### CODE REVIEW

File: insonmnia/hub/eth.go [hunk 1]
```diff
@@ -24,7 +24,7 @@ type ETH interface {
 	AcceptDeal(id string) error
 
 	// CheckDealExists checks whether a given deal exists.
-	CheckDealExists(id string) (bool, error)
+	CheckDealExists(id string) (*pb.Deal, error)
```
Comment (oopcode) [inline insonmnia/hub/eth.go]: This looks more like `GetDeal`, doesn't it?

Comment (sshaman1101) [reply insonmnia/hub/eth.go]: Done!

File: insonmnia/hub/server.go [hunk 2]
```diff
@@ -432,16 +434,22 @@ func (h *Hub) generateTaskID() string {
...
+	meta, ok := h.deals[DealID(deal.GetId())]
+	h.tasksMu.Unlock()
+
+	if !ok {
+		// Hub knows nothing about this deal
+		return nil, errDealNotFound
 	}
 
 	// Extract proper miner associated with the deal specified.
-	miner, usage, err := h.findMinerByOrder(OrderId(request.GetDealId()))
```
Comment (oopcode) [inline insonmnia/hub/server.go]: Shouldn't we always use deal ID and never use order ID?

**Comment by oopcode** [comment]
LGTM, except that is doesn't build.

**Comment by sshaman1101** [comment]
Fixed!

**Comment by oopcode** [comment]
LGTM, lol.

**Pull request merged**